diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/FileStructureRule.kt

```diff
@@ -263,21 +263,23 @@ class FileStructureRule(private val configRules: List<RulesConfig>) : Rule("file
         else -> (headerKdoc ?: copyrightComment) to firstCodeNode
     }
 
-    @Suppress("TYPE_ALIAS")
+    @Suppress("TYPE_ALIAS", "UnsafeCallOnNullableType")
     private fun regroupImports(imports: List<KtImportDirective>): List<List<KtImportDirective>> {
         val (android, notAndroid) = imports.partition {
             it.isStandard(StandardPlatforms.ANDROID)
         }
 
-        val (ownDomain, tmp) = notAndroid.partition { import ->
-            import
-                .importPath
-                ?.fqName
-                ?.pathSegments()
-                ?.zip(domainName.split(PACKAGE_SEPARATOR).map(Name::identifier))
-                ?.all { it.first == it.second }
-                ?: false
-        }
+        val (ownDomain, tmp) = domainName?.let {
+            notAndroid.partition { import ->
+                import
+                    .importPath
+                    ?.fqName
+                    ?.pathSegments()
+                    ?.zip(domainName!!.split(PACKAGE_SEPARATOR).map(Name::identifier))
```

Comment:
Isn't `domainName` smartcasted to non-nullable type here?

Reply:
Yes, it is